### PR TITLE
Enable Automated UI Testing

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): #

## PR Type
What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/articles/uno-development/working-with-the-samples-apps.md [hunk 1]
```diff
@@ -21,6 +21,16 @@ To create a UI Test for the sample applications:
 - Add `[Uno.UI.Samples.Controls.SampleControlInfo("Replace_with_control_or_class_name", "MyTestName", description: "MyDescription")]` on the code-behind class.
 - Run the samples application, and the sample should appear in the samples browser
 
+The Uno UI process validates does two types of validations:
```
Comment (davidjohnoliver) [inline doc/articles/uno-development/working-with-the-samples-apps.md]:
```suggestion
The Uno.UI build process does two types of validations:
```

File: src/SamplesApp/SamplesApp.UITests/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+﻿Please follow [this guide](doc/articles/uno-development/working-with-the-samples-apps.md) to run and author automated UI Tests.
```
Comment (davidjohnoliver) [inline src/SamplesApp/SamplesApp.UITests/README.md]: just eyeballing the link, isn't it missing a `../`, or multiples?

File: src/SamplesApp/UITests.Shared/UITests.Shared.projitems [hunk 3]
```diff
@@ -3530,6 +3685,96 @@
     <Compile Update="$(MSBuildThisFileDirectory)Windows_UI_Xaml_Controls\ToggleSwitchControl\ToggleSwitch_IsOn.xaml.cs">
       <DependentUpon>ToggleSwitch_IsOn.xaml</DependentUpon>
     </Compile>
+    <Compile Update="C:\s\nv.github\uno\src\SamplesApp\UITests.Shared\Windows_UI_Xaml\TouchEventsTests\TappedEventTest.xaml.cs">
```
Comment (davidjohnoliver) [inline src/SamplesApp/UITests.Shared/UITests.Shared.projitems]: Lots of absolute paths here and below

File: src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.iOS.tt [hunk 4]
```diff
@@ -30,8 +30,10 @@ using Windows.UI.Xaml.Automation;
 
 namespace <#= mixin.NamespaceName #>
 {
-    public partial class <#= mixin.ClassName #> : IFrameworkElement
+    public partial class <#= mixin.ClassName #> : IFrameworkElement, IXUidProvider
 	{		
+		string IXUidProvider.Uid { get; set; }
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.iOS.tt]: should this only be for non-`FrameworkElements`?

File: src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.Android.tt [hunk 5]
```diff
@@ -49,6 +49,8 @@ namespace <#= mixin.NamespaceName #>
 
 		public event SizeChangedEventHandler SizeChanged;
 
+		string IXUidProvider.Uid { get; set; }
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.Android.tt]: should this only be for non-`FrameworkElements`?

File: src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.macOS.tt [hunk 6]
```diff
@@ -30,8 +30,10 @@ using Windows.UI.Xaml.Automation;
 
 namespace <#= mixin.NamespaceName #>
 {
-	public partial class <#= mixin.ClassName #> : IFrameworkElement
+	public partial class <#= mixin.ClassName #> : IFrameworkElement, IXUidProvider
 	{		
+		string IXUidProvider.Uid { get; set; }
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.macOS.tt]: should this only be for non-`FrameworkElements`?

File: src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.iOS.tt (hunk 4 above)
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.iOS.tt]: It avoids making two checks in the SetXName helper. Once we remove IFrameworkElement, we can remove this.

**Pull request merged**